### PR TITLE
build jbmc on AWS codebuild

### PR DESCRIPTION
**Pull request by kroening**

### CODE REVIEW

File: buildspec.yml [hunk 1]
```diff
@@ -3,22 +3,26 @@ version: 0.2
...
-      - (cd src ; make minisat2-download)
-      - (cd src ; make CXX="ccache g++" -j2)
-      - (cd regression ; make test)
+      - make -C src CXX="ccache g++" -j2
+      - make -C unit CXX="ccache g++" -j2
+      - make -C jbmc/src CXX="ccache g++" -j2
+      - make -C jbmc/unit CXX="ccache g++" -j2
   post_build:
     commands:
+      - make -C regression test
```
Comment (smowton) [inline buildspec.yml]: jbmc regression tests too?

File: buildspec.yml [hunk 2]
```diff
@@ -3,22 +3,26 @@ version: 0.2
...
   build:
     commands:
       - echo Build started on `date`
+      - make -C src minisat2-download
       - make -C jbmc/src java-models-library-download
-      - (cd src ; make minisat2-download)
-      - (cd src ; make CXX="ccache g++" -j2)
-      - (cd regression ; make test)
+      - make -C src CXX="ccache g++" -j2
+      - make -C unit CXX="ccache g++" -j2
```
Comment (smowton) [inline buildspec.yml:20]: AFAIK this is building the unit tests but not running them?

File: buildspec.yml [hunk 3]
```diff
@@ -3,22 +3,29 @@ version: 0.2
 phases:
   install:
     commands:
-      - echo "deb http://ppa.launchpad.net/ubuntu-toolchain-r/test/ubuntu trusty main" > /etc/apt/sources.list.d/toolchain.list
+      - add-apt-repository ppa:ubuntu-toolchain-r/test -y
```
Comment (tautschnig) [inline buildspec.yml:6]: Is `add-apt-repository` natively available, without any further installation steps?

File: buildspec.yml (hunk 2 above)
Comment (tautschnig) [reply buildspec.yml:20]: For the record: It seems "running the test" has been added as `post_build` steps.

**Review by tautschnig — COMMENTED** [review]
Looks ok to me, just a question below.

**Review by tautschnig — APPROVED** [review]
(Sorry, meant to select "Approve" in my review moments ago.)

**Review by peterschrammel — APPROVED** [review]

File: buildspec.yml (hunk 3 above)
Comment (kroening) [reply buildspec.yml:6]: Yes; furthermore, this is already the case on the underlying Ubuntu trusty; it's not a tweak in the docker image.

**Comment by kroening** [comment]
BTW, just spotted that there is an AWS-provided image that already comes with jdk8, saving some nontrivial download time.

**Pull request merged**